app/src/main/java/com/project/app/base/BaseFragment.kt

```diff
@@ -0,0 +1,158 @@
+package com.project.app.base
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Bundle
+import android.view.*
+import android.view.inputmethod.InputMethodManager
+import android.widget.ImageButton
+import android.widget.TextView
+import android.widget.Toast
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.findNavController
+import androidx.viewbinding.ViewBinding
+import com.project.app.R
+import com.project.app.helpers.LoadingDialog
+import com.project.app.helpers.ToolbarType
+import kotlinx.coroutines.Job
+
+abstract class BaseFragment<VB : ViewBinding> : Fragment(), IBaseFragment {
+    lateinit var activityIndicator: LoadingDialog
+    private var job = Job()
+    private lateinit var topBarLayout: ConstraintLayout
+    lateinit var leftButtonToolbar: ImageButton
+    private lateinit var textViewMenuTitle: TextView
+    private lateinit var rigthButtonToolbar: ImageButton
+
+    private lateinit var fragmentView: View
+
+    private var _binding: ViewBinding? = null
+    abstract val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> VB
+    @Suppress("UNCHECKED_CAST")
+    val binding: VB
+        get() = _binding as VB
```

Comment:
Para evitar tener este `UNCHECKED_CAST` en el gey del binding, no podremos hacer que `_binding` sea de tipo `VB`? Es decir...

```suggestion
    private var _binding: VB? = null
    abstract val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> VB
    val binding: VB
        get() = _binding
```

De esta forma, es un poco más seguro. Nos evitamos tener que usar la annotation por errores de compilación al castear.

Más allá de eso, habrá que ver si está bueno tener un `binding` y un `_binding`. La diferencia que veo es que uno no es nullable. Quizás esto lo podamos simplificar un poco.

Quizás podamos solo quedarnos con `binding` y que este sea un lateinit var si lo vemos necesario.

Reply:
Lo pruebo y te lo confirmo, realmente es algo nuevo (parte de la migración que hice) y no sabría responder con seguridad hablando del tema del cast

Reply:
Oki. Yo también hablo desde el desconocimiento. Si querés, pasame los ejemplos que viste y veo si se me ocurre algo. También es algo que podemos ver cuando nos juntemos nuevamentwe.